rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -19,164 +19,362 @@
-        );
-        Federation newFederation = FederationTestUtils.getErpFederation(
-            networkParameters,
-            newFederationPublicKeys,
-            newFedCreationBlockNumber
-        );
+        @Test
+        void getActiveFederationRedeemScript_beforeRSKIP293_returnsEmpty() {
+            ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+            when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(false);
+
+            federationSupport = federationSupportBuilder
+                .withFederationConstants(federationMainnetConstants)
+                .withFederationStorageProvider(storageProvider)
+                .withActivations(activations)
+                .build();
+            assertFalse(federationSupport.getActiveFederationRedeemScript().isPresent());
+        }
 
-        List<BtcECKey> oldFederationPublicKeys = BitcoinTestUtils.getBtcEcKeysFromSeeds(
-            new String[]{"fa10", "fa12", "fa13", "fa14", "fa15", "fa16", "fa17"}, true
-        );
-        Federation oldFederation = FederationTestUtils.getErpFederation(
-            networkParameters,
-            oldFederationPublicKeys,
-            oldFedCreationBlockNumber
-        );
+        @ParameterizedTest
+        @MethodSource("expectedRedeemScript_args")
+        void getActiveFederationRedeemScript_returnsExpectedRedeemScriptAccordingToActivationAgeAndActivations(long currentBlock, ActivationConfig.ForBlock activations, Script expectedRedeemScript) {
+            Block executionBlock = mock(Block.class);
+            when(executionBlock.getNumber()).thenReturn(currentBlock);
+
+            federationSupport = federationSupportBuilder
+                .withFederationConstants(federationMainnetConstants)
+                .withFederationStorageProvider(storageProvider)
+                .withRskExecutionBlock(executionBlock)
+                .withActivations(activations)
+                .build();
+
+            Optional<Script> activeFederationRedeemScript = federationSupport.getActiveFederationRedeemScript();
+            assertTrue(activeFederationRedeemScript.isPresent());
+            assertThat(activeFederationRedeemScript.get(), is(expectedRedeemScript));
+        }
 
-        storageProvider.setNewFederation(newFederation);
-        storageProvider.setOldFederation(oldFederation);
+        private Stream<Arguments> expectedRedeemScript_args() {
+            return Stream.of(
+                Arguments.of(blockNumberFederationActivationHop - 1, hopActivations, oldFederation.getRedeemScript()), // new federation shouldn't be active
+                Arguments.of(blockNumberFederationActivationHop, hopActivations, newFederation.getRedeemScript()), // new federation should be active
+                Arguments.of(blockNumberFederationActivationHop, fingerrootActivations, oldFederation.getRedeemScript()), // new federation shouldn't be active since hop activation block number is smaller than fingerroot one
```

Comment:
nit: since this comment is very long, consider putting it above instead of next to it